src/Extensions/ElementalAreasExtension.php

```diff
@@ -223,24 +223,26 @@ protected function onBeforeWrite()
         if (!$this->supportsElemental()) {
             return;
         }
-
-        $elementalAreaRelations = $this->owner->getElementalRelations();
-
+        $owner = $this->getOwner();
+        $elementalAreaRelations = $owner->getElementalRelations();
         $this->ensureElementalAreasExist($elementalAreaRelations);
-
-        $ownerClassName = get_class($this->owner);
+        $ownerClassName = $owner->ClassName;
```

Comment:
The `ClassName` might _not actually match_ `get_class` if:
1. Injector has replaced the class, or
2. `ClassName` is explicitly intentionally changed

BUT in both of those cases we want `ClassName` anyway so this is actually an unintended bug fix.